lessons/4.mdx

```diff
@@ -0,0 +1,818 @@
+---
+title: Testing your Tier NFT
+description:
+  Prove your Tier NFT contract works with automated tests! Builds on knowledge
+  from Lesson 3.
+icons: ['solidity', 'hardhat', 'openzeppelin']
+---
+
+# Lesson 4: Testing your Tier NFT
+
+In this lesson, we'll be adding automated tests for our smart contract created
+in Lesson 3.
+
+Here you'll:
+
+- Learn about the benefits of writing automated tests, general testing concepts
+  and best practices
+- Walk away with many practical tests for smart contracts
+- Nibble on some food for thought about test-driven development
+
+## Why automated testing?
+
+Testing can:
+
+- prove code actually works
+- help you write more modular code
+- ensure code doesn't break when you or others make changes or add new features.
+  People new to the code-base may not even know when something breaks without
+  tests.
+- ensure a bug is fixed and won't come back again
+- give you a safety net when you refactor code
+
+Tests are a tool for writing your code and making sure it keeps working as
+expected when changes are made.
+
+Automated tests don't fully replace manually testing your smart contract but
+it's an important tool to ensuring your contract will work as expected.
+
+## Let's add some tests
+
+### First add a test framework
+
+You've already setup hardhat in Lesson 2, which is a great framework for writing
```

Comment:
We are doing an **almighty jump* from here to line 46.....

Reply:
Closing comment, solving in related follow-up comment that's discussing signposts to Lesson 2.